src/lang-promql/parser/matcher.ts

```diff
@@ -80,3 +58,40 @@ export function buildLabelMatchers(labelMatchers: SyntaxNode[], state: EditorSta
   });
   return matchers;
 }
+
+export function labelMatchersToString(metricName: string, matchers?: Matcher[]): string {
+  if (!matchers || matchers.length === 0) {
+    return metricName;
+  }
+
+  let matchersAsString = '';
+  for (const matcher of matchers) {
+    if (matcher.value === '') {
```

Comment:
This looks great 😄  One thing, because we have the label name we are trying to match, would it be safer to pass that value into this function and make this `if (matcher.name === labelName || matcher.value === '')`.  If the user initiates autocomplete manually by using the keyboard shortcut, the current way of checking if the value is the empty string would not work as expected (if they have already typed something in the parentheses).

Reply:
ah yeah indeed. Thanks I will change the logic

Reply:
mmm actually that's a different flow. Here I only did it for the autocompletion of the labelValue. Not sure it makes sense to add the matcher in the labelName prometheus method

Reply:
Yeah, I agree that it doesn't make sense to add this for the labelName method :) An example of what I was saying above is the user types `rate{labelName="lab|"}` and then manually initiates autocomplete.  In this case, matcher.value will not be the empty string, meaning that we will incorrectly include the `labelName` matcher in the request to /series.  However, if we check that `matcher.name === labelName` we would still correctly remove it from the results vector selector.

Reply:
Ah ok. Thanks for the explenation. Makes sense now :). I will make the changes